src/generator/gomod.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Session } from '@azure-tools/autorest-extension-base';
+import { CodeModel } from '@azure-tools/codemodel';
+
+// Creates the content in go.mod if the --module switch was specified
+export async function generateGoModFile(session: Session<CodeModel>): Promise<string> {
+  const modName = await session.getValue('module', 'none');
+  if (modName === 'none') {
+    return '';
+  }
+  let text = `module ${modName}\n\n`;
+  text += 'go 1.13\n\n';
```

Comment:
At some future point we should also think about what to do when we want to update go versions.